lib/puppet/provider/ini_setting/ruby.rb

```diff
@@ -35,7 +35,14 @@ def self.namevar(section_name, setting)
   end
 
   def exists?
-    !ini_file.get_value(section, setting).nil?
+    if ini_file.section?(section)
+      !ini_file.get_value(section, setting).nil?
+    elsif !resource[:force_new_section_creation]
```

Comment:
So the issue i'm seeing is with this. If you're child object doesn't have this parameter (none of ours do because we're not inheirting from the upstram ini_setting type) this fails. I think what we need to do is check that this option is a value on the type before trying to consume it.